src/components/Header/NavComponents/userprofile.tsx

```diff
@@ -57,18 +65,21 @@ export const UserProfile: React.FC<{}> = () => {
         bg={mode({ base: "secondary.800", md: "primary.500" }, "primary.500")}
         rounded="lg"
         px="5px"
+        mr="2px"
       >
         <Text mr="5px">{userBal}</Text>
         <Text fontWeight="400">AGVE</Text>
       </Center>
 
+      {connector ? isMetamask ? <ChainSelector /> : <CurrentChainBox /> : <></>}
```

Comment:
You can make this more efficient at rendering, and less likely to drop the content of form inputs by memoizing:

```
const chainSelector = React.useMemo(
  () => (connector ? isMetamask ? <ChainSelector /> : <CurrentChainBox /> : null),
  [connector, isMetamask]);
```

Then replace line 74 (`{connector ? isMetamask ? <ChainSelector /> : <CurrentChainBox /> : <></>}`) with `{chainSelector}`.

Reply:
Did this myself in the later commit, check it out and resolve this comment when you see it.